jobs/import_delete/worker.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package import_delete
+
+import (
+	"errors"
+	"path/filepath"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/jobs"
+	tjobs "github.com/mattermost/mattermost-server/v5/jobs/interfaces"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+)
+
+func init() {
+	app.RegisterJobsImportDeleteInterface(func(a *app.App) tjobs.ImportDeleteInterface {
+		return &ImportDeleteInterfaceImpl{a}
+	})
+}
+
+type ImportDeleteInterfaceImpl struct {
+	app *app.App
+}
+
+type ImportDeleteWorker struct {
+	name        string
+	stopChan    chan struct{}
+	stoppedChan chan struct{}
+	jobsChan    chan model.Job
+	jobServer   *jobs.JobServer
+	app         *app.App
+}
+
+func (i *ImportDeleteInterfaceImpl) MakeWorker() model.Worker {
+	return &ImportDeleteWorker{
+		name:        "ImportDelete",
+		stopChan:    make(chan struct{}),
+		stoppedChan: make(chan struct{}),
+		jobsChan:    make(chan model.Job),
+		jobServer:   i.app.Srv().Jobs,
+		app:         i.app,
+	}
+}
+
+func (w *ImportDeleteWorker) JobChannel() chan<- model.Job {
+	return w.jobsChan
+}
+
+func (w *ImportDeleteWorker) Run() {
+	mlog.Debug("Worker started", mlog.String("worker", w.name))
+
+	defer func() {
+		mlog.Debug("Worker finished", mlog.String("worker", w.name))
+		close(w.stoppedChan)
+	}()
+
+	for {
+		select {
+		case <-w.stopChan:
+			mlog.Debug("Worker received stop signal", mlog.String("worker", w.name))
+			return
+		case job := <-w.jobsChan:
+			mlog.Debug("Worker received a new candidate job.", mlog.String("worker", w.name))
+			w.doJob(&job)
+		}
+	}
+}
+
+func (w *ImportDeleteWorker) Stop() {
+	mlog.Debug("Worker stopping", mlog.String("worker", w.name))
+	close(w.stopChan)
+	<-w.stoppedChan
+}
+
+func (w *ImportDeleteWorker) doJob(job *model.Job) {
+	if claimed, err := w.jobServer.ClaimJob(job); err != nil {
+		mlog.Warn("Worker experienced an error while trying to claim job",
+			mlog.String("worker", w.name),
+			mlog.String("job_id", job.Id),
+			mlog.String("error", err.Error()))
+		return
+	} else if !claimed {
+		return
+	}
+
+	importPath := *w.app.Config().ImportSettings.Directory
+	retentionTime := time.Duration(*w.app.Config().ImportSettings.RetentionDays) * 24 * time.Hour
+	imports, appErr := w.app.ListDirectory(importPath)
+	if appErr != nil {
+		w.setJobError(job, appErr)
+		return
+	}
+
+	for i := range imports {
+		filename := filepath.Base(imports[i])
+		modTime, appErr := w.app.FileModTime(filepath.Join(importPath, filename))
+		if appErr != nil {
+			mlog.Error("Worker: Failed to get file modification time", mlog.Err(appErr))
```

Comment:
I understand this is to keep the code consistent. But these are probably subject to change. We should use a mechanism to collate the errors and print as a _warning_ in the end since we eventually consider the job is finished successfully. Using `DEBUG` level here would be more convenient IMHO. What do you think?

Reply:
Make sense :+1: